soc/nxp/imxrt/imxrt118x/soc.h

```diff
@@ -20,6 +20,10 @@
 extern "C" {
 #endif
 
+#ifdef CONFIG_MEMC
```

Comment:
```suggestion
#ifdef CONFIG_MEMC_MCUX_FLEXSPI
```

Let's use the same dependency between the CMake file where we pull in the C code for this function, and the function prototype

Reply:
Updated.